matlab/schemas/+tuning/DotRF.m

```diff
@@ -0,0 +1,206 @@
+%{
+tuning.DotRF (computed) # receptive field from the dot stimulus
+-> preprocess.Sync
+-> preprocess.Spikes
+-> tuning.DotRFMethod
+-----
+%}
+
+classdef DotRF < dj.Relvar & dj.AutoPopulate
+    
+    properties
+        popRel = tuning.DotRFMethod * preprocess.Spikes & ...
+            (preprocess.Sync & (vis.ScanConditions & vis.SingleDot))
+    end
+    
+    methods(Access=protected)
+        
+        function makeTuples(self, key)
+            % get parameters
+            [mon_size, mon_dist] = fetch1(vis.Session &...
+                (preprocess.Spikes * preprocess.Sync & vis.SingleDot & key), ...
+                'monitor_size','monitor_distance');
+            [onset_delay, response_duration, rf_filter, shuffle, sd] = ...
+                fetch1(tuning.DotRFMethod & key, ...
+                'onset_delay', 'response_duration', 'rf_filter', 'shuffle', 'rf_sd');
+            
+            % get stimulus conditions
+            trials = fetch((preprocess.Sync & key) * vis.Trial * vis.SingleDot  ...
```

Comment:
Due to MySQL quirks, very likely, moving all the restrictions to the end will produce a faster query:
```matlab
trials = fetch(preprocess.Sync * vis.Trial * vis.SingleDot ...
    & key & 'trial_idx between first_trial and last_trial');
```